dev-tools/mage/check.go

```diff
@@ -270,13 +270,14 @@ type dashboardObject struct {
 		Title                 string `json:"title"`
 		KibanaSavedObjectMeta *struct {
 			SearchSourceJSON struct {
-				Index string `json:"index"`
+				Index *string `json:"index"`
```

Comment:
why do we need a pointer here?

Reply:
I added a pointer here because I need to distinguish if this field exists or the value of this field is empty string `""`. Without the pointer, even if `VisState.Params` doesn't have `IndexPattern` field, it will return `""`.